src/contracts/adapters/linea/LineaAdapter.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import {IMessageService} from './interfaces/IMessageService.sol';
+import {BaseAdapter, IBaseAdapter} from '../BaseAdapter.sol';
+import {ChainIds} from 'solidity-utils/contracts/utils/ChainHelpers.sol';
+import {Errors} from '../../libs/Errors.sol';
+import {ILineaAdapter} from './ILineaAdapter.sol';
+import {SafeCast} from 'solidity-utils/contracts/oz-common/SafeCast.sol';
+
+/**
+ * @title LineaAdapter
+ * @author BGD Labs
+ * @notice Optimism bridge adapter. Used to send and receive messages cross chain between Ethereum and Optimism
+ * @dev it uses the eth balance of CrossChainController contract to pay for message bridging as the method to bridge
+        is called via delegate call
+ * @dev note that this adapter is can only be used for the communication path ETHEREUM -> LINEA
+ */
+contract LineaAdapter is ILineaAdapter, BaseAdapter {
+  /// @inheritdoc ILineaAdapter
+  address public immutable LINEA_MESSAGE_SERVICE;
+
+  /**
+   * @notice only calls from the set ovm are accepted.
+   */
+  modifier onlyLineaMessageService() {
```

Comment:
this modifier used only once, not sure that needed

Reply:
I prefer to leave it like this, as it follows the same design as all other bridges, and dont think it adds significant extra gas